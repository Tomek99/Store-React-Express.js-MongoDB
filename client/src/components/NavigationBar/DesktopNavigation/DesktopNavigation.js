import React from "react";
import NavListElement from "../NavListElement/NavListElement";
import styles from "./DesktopNavigation.module.scss";
import navigationBarList from "../../../data/navigationBarList.json";
function DesktopNavigation() {
  return (
    <ul className={styles.DesktopNavigation}>
      {navigationBarList.map((item, index) => (
        <NavListElement
          isLink={false}
          key={index}
          name={item.name}
          path={item.path}
        />
      ))}
    </ul>
  );
}

export default DesktopNavigation;
