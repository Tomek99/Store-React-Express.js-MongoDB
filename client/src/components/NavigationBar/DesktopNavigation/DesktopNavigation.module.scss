.DesktopNavigation {
    display: flex;
    flex-direction: row;
    align-items: center;
    list-style-type: none;
    gap: 20px;
    font-size: 2rem;
}

@media (max-width: 900px) {
    .DesktopNavigation {
        display: none !important;
    }
}
