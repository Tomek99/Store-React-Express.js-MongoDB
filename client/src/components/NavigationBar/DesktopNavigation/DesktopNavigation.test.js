import { render } from "@testing-library/react";
import DesktopNavigation from "./DesktopNavigation";
import { MemoryRouter } from "react-router-dom";

test("should render DesktopNavigation component", () => {
  const { container } = render(
    <MemoryRouter>
      <DesktopNavigation />
    </MemoryRouter>
  );
  expect(container).toMatchSnapshot();
});
