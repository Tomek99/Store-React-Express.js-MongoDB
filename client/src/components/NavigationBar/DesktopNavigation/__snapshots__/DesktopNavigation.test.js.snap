// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render DesktopNavigation component 1`] = `
<div>
  <ul
    class="DesktopNavigation"
  >
    <li
      class="NavListElement"
    >
      <a
        aria-current="page"
        class="activeNavLink"
        href="/"
      >
        Home
      </a>
    </li>
    <li
      class="NavListElement"
    >
      <a
        class="navLink"
        href="/products"
      >
        Products
      </a>
    </li>
    <li
      class="NavListElement"
    >
      <a
        class="navLink"
        href="/about-us"
      >
        About
      </a>
    </li>
    <li
      class="NavListElement"
    >
      <a
        class="navLink"
        href="/contact"
      >
        Contact
      </a>
    </li>
    <li
      class="NavListElement"
    >
      <a
        class="navLink"
        href="/blog"
      >
        Blog
      </a>
    </li>
  </ul>
</div>
`;
